src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseConcreteRequestResolver.java

```diff
@@ -0,0 +1,57 @@
+package com.minecolonies.coremod.colony.requestsystem.resolvers;
+
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.manager.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requestable.IConcreteDeliverable;
+import com.minecolonies.api.colony.requestsystem.requestable.IDeliverable;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingWareHouse;
+import com.minecolonies.coremod.colony.requestsystem.requesters.BuildingBasedRequester;
+import com.minecolonies.coremod.colony.requestsystem.resolvers.core.AbstractWarehouseRequestResolver;
+import com.minecolonies.coremod.tileentities.TileEntityWareHouse;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * ----------------------- Not Documented Object ---------------------
+ */
+public class WarehouseConcreteRequestResolver extends AbstractWarehouseRequestResolver
```

Comment:
The factory for this class is missing.

Reply:
And, this is why the code you went "wtf" about. I didn't realize that there was a factory pattern set up for the WarehouseRequestResolver, as it doesn't actually appear to be in use.  I'll gladly add the factory and remove the  newInstanceOfSelf.

Reply:
The entire RS uses factories that have to be registered. They handle the entire IO for it.

Reply:
So, apparently I figured it out, added it, and forgot. The factories actually are there, but with the formatting issue in Stack.java it was not apparent.